### PR TITLE
a few fixes

### PR DESCRIPTION
**Pull request by cbielow**
# Description

mostly from failing nightly builds and some recently introduced which were not caught by CI

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.
- Or click on the action: e.g., for clang-format linting

# Note:
- Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

# Advanced commands (admins / reviewer only):
- /rebase will try to rebase the PR on the current develop branch.
- /reformat (experimental) applies the clang-format style changes as additional commit
- setting the label NoJenkins will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)

### CODE REVIEW

**Review by jpfeuffer — APPROVED** [review]

File: src/openms/source/ANALYSIS/ID/AhoCorasickAmbiguous.cpp [hunk 1]
```diff
@@ -278,8 +278,9 @@ namespace OpenMS
       {
         const auto needle_length = trie_[i()].depth_and_hits.depth;
         const auto text_start = text_pos - needle_length;
-        for (const auto needle_idx : umap_index2needles_.at(i())) {
-          hits.emplace_back(needle_idx, needle_length, text_start);
+        for (const auto needle_idx : umap_index2needles_.at(i()))
+        {
+          hits.emplace_back(needle_idx, needle_length, Hit::T(text_start));
```
Comment (timosachsenberg) [inline src/openms/source/ANALYSIS/ID/AhoCorasickAmbiguous.cpp:283]: constructor? conversion?

**Comment by timosachsenberg** [comment]
http://cdash.openms.de/viewBuildError.php?buildid=232264

File: src/openms/source/ANALYSIS/ID/AhoCorasickAmbiguous.cpp (hunk 1 above)
Comment (cbielow) [reply src/openms/source/ANALYSIS/ID/AhoCorasickAmbiguous.cpp:283]: conversion at the moment.
I wanted to force users to make a conscious decision about it.

**Comment by cbielow** [comment]
tests pass.
Any more comments?

**Pull request merged**